### PR TITLE
fix(STONEINTG-598): clarify required tests

### PR DESCRIPTION
**Pull request by MartinBasti**
All tests are required to pass by default. Explain that marking scenario optional bypass this default setting.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc [hunk 1]
```diff
@@ -18,7 +18,9 @@ Complete the following steps in the {ProductName} console:
 . Optional: If you need to deploy your application as part of the test, use the drop-down menu in the *Environment* field to select *development*. {ProductName} then deploys your application to a temporary clone of that environment during testing.
 +
 WARNING: Without specifying an Environment, an integration test can only do static testing. To make application endpoints available for testing, you must deploy your application to an ephemeral environment.
-. Optional: If you do not want the test to prevent the application from being deployed, then select *Mark as optional for release*.
+. Optional: *Mark as optional for release* if you want to allow this test scenario to fail without affecting deployment and release of the application.
++
+NOTE: By default all integration test scenarios are required to pass. A failing integration test marks the application snapshot as failed and prevents deploying and releasing of the application. By marking integration test scenario as optional this test will not prevent the application snapshot to be deployed and released even if it fails.
```
Comment (Josh-Everett) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc:23]: LGTM - some useful context here

**Review by Josh-Everett — APPROVED** [review]

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Review by sonam1412 — APPROVED** [review]

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc [hunk 2]
```diff
@@ -1,6 +1,6 @@
 = Adding an integration test
 
-You can add an integration test to ensure that individual components correctly integrate to form a whole application. {ProductName} runs integration tests on the container images of components before it deploys those images.     
+You can add an integration test to ensure that individual components correctly integrate to form a whole application. {ProductName} runs integration tests on the container images of components before it deploys those images.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc:3]:
```suggestion
in {ProductName}, you can add integration tests to verify that the individual components of your application integrate correctly, forming a complete and functional application. {ProductName} runs these integration tests on the container images of components before their deployment.
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc [hunk 3]
```diff
@@ -18,7 +18,9 @@ Complete the following steps in the {ProductName} console:
 . Optional: If you need to deploy your application as part of the test, use the drop-down menu in the *Environment* field to select *development*. {ProductName} then deploys your application to a temporary clone of that environment during testing.
 +
 WARNING: Without specifying an Environment, an integration test can only do static testing. To make application endpoints available for testing, you must deploy your application to an ephemeral environment.
-. Optional: If you do not want the test to prevent the application from being deployed, then select *Mark as optional for release*.
+. Optional: *Mark as optional for release* if you want to allow this test scenario to fail without affecting deployment and release of the application.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc:21]:
```suggestion
. Optional: You can choose to select *Mark as optional for release*. Doing so allows the integration tests to fail without impacting the deployment and release process of your application. 
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc (hunk 1 above)
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc:23]:
```suggestion
NOTE: By default, all integration test scenarios are mandatory and must pass. A failing integration test marks the application snapshot as failed, preventing its deployment and release. However,  if you have selected *Mark as optional for release*, a failure in this test won't hinder the deployment and release of the application snapshot.
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc (hunk 2 above)
Comment (MartinBasti) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc:3]: actually I didn't change this, I only accidentally pushed whitespace removal done by IDE. But can include it, no problem

**Comment by MartinBasti** [comment]
I update PR with requested changes, thank you!

**Review by dirgim — APPROVED** [review]
LGTM

**Review by gtrivedi88 — APPROVED** [review]

**Pull request merged**